engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/ProcessDefinitionManager.java

```diff
@@ -164,9 +165,8 @@ protected ProcessDefinitionEntity findProcessDefinitionByKeyVersionOrVersionTag(
   }
 
   public List<ProcessDefinition> findProcessDefinitionsByKey(String processDefinitionKey) {
-    ProcessDefinitionQueryImpl processDefinitionQuery = new ProcessDefinitionQueryImpl()
-      .processDefinitionKey(processDefinitionKey);
-    return  findProcessDefinitionsByQueryCriteria(processDefinitionQuery, null);
+    ProcessDefinitionQueryImpl processDefinitionQuery = new ProcessDefinitionQueryImpl().processDefinitionKeyAsImpl(processDefinitionKey);
+    return findProcessDefinitionsByQueryCriteria(processDefinitionQuery, null);
```

Comment:
This change will be avoided if return types are left as they are in the implementation class.

Reply:
I fixed that